connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -309,18 +309,37 @@ private void createOffsetSyncsTopic() {
     // visible for testing
     void computeAndCreateTopicPartitions()
             throws InterruptedException, ExecutionException {
-        Map<String, Long> partitionCounts = knownSourceTopicPartitions.stream()
+        Map<String, Long> topicToParitionCount = knownSourceTopicPartitions.stream()
             .collect(Collectors.groupingBy(TopicPartition::topic, Collectors.counting())).entrySet().stream()
             .collect(Collectors.toMap(x -> formatRemoteTopic(x.getKey()), Entry::getValue));
         Set<String> knownTargetTopics = toTopics(knownTargetTopicPartitions);
-        List<NewTopic> newTopics = partitionCounts.entrySet().stream()
-            .filter(x -> !knownTargetTopics.contains(x.getKey()))
-            .map(x -> new NewTopic(x.getKey(), x.getValue().intValue(), (short) replicationFactor))
+
+        // get the set of topics that are not present on the target
+        Set<String> topicsToCreate = topicToParitionCount.keySet().stream()
+                .filter(topic -> !knownTargetTopics.contains(topic))
+                .collect(Collectors.toSet());
+        if (topicsToCreate.isEmpty())
+            return;
+
+        // get corresponding topic configurations from the source
+        Map<String, Config> configs = describeTopicConfigs(topicsToCreate);
```

Comment:
This is not the right way to grab the topic configs as the names have been formatted using `formatRemoteTopic`.

Reply:
Addressed in db09806.